feelpp/feel/feelmesh/geoentity.hpp

```diff
@@ -769,6 +772,147 @@ class GeoEntity
 
     //@}
 
+    /**
+     * set the tags associated to the points
+     * - tags[0] physical region
+     * - tags[1] elementary region
+     * - tags[2] particular region
+     */
+    virtual void setTags( std::vector<int> const& tags )
+    {
+        M_markers[1].assign( tags[0] );
+        if ( tags.size() > 1 )
+            M_markers[2].assign( tags[1] );
+
+        if ( tags.size() > 2 )
+        {
+            this->setProcessId( tags[3] );
+
+            if ( tags[2] > 1 )
+            {
+                // ghosts
+                std::vector<rank_type> p( tags[2] - 1 );
+
+                for ( size_type i = 0; i < p.size(); ++i )
+                {
+                    p[i] = tags[4 + i];
+                }
+
+                this->setNeighborPartitionIds( p );
+            }
+        }
+    }
+
+    //! return all markers
+    std::map<uint16_type, marker_type> const&
+    markers() const
+    {
+        return M_markers;
+    }
+    //! set all markers
+    void setMarkers( std::map<uint16_type, marker_type> const& markers )
+    {
+        M_markers = markers;
+    }
+
+    //! return true if has marker type id k
+    bool hasMarkerType( uint16_type k ) const
+    {
+        auto itFindMarker = M_markers.find( k );
+        if ( itFindMarker == M_markers.end() )
+            return false;
+        if ( itFindMarker->second.isOff() )
+            return false;
+        return true;
+    }
+    //! return true if has marker type id k
+    bool hasMarker( uint16_type k ) const { return this->hasMarkerType( k ); }
+
+    //! return marker with type id k
+    marker_type const& marker( uint16_type k ) const
+    {
+        DCHECK( this->hasMarkerType( k ) ) << "no marker type " << k;
+        return M_markers.find( k )->second;
+    }
+    //! set marker type id k with flag v
+    void setMarker( uint16_type k, flag_type v )
+    {
+        M_markers[k].assign( v );
+    }
+    //! add flag v to the marker type id k
+    void addMarker( uint16_type k, flag_type v )
+    {
+        M_markers[k].insert( v );
+    }
+    //! set marker type id k with flag v
+    template <typename TT,std::enable_if_t< is_iterable_of_v<TT,flag_type>, bool> = true >
+    void setMarker( uint16_type k,TT const& vs )
+    {
+        M_markers[k].assign( vs );
+    }
+    //! add flags vs to the marker type id k
+    template <typename TT,std::enable_if_t< is_iterable_of_v<TT,flag_type>, bool> = true >
+    void addMarker( uint16_type k, TT const& vs )
```

Comment:
candidate for constexpr and merge with the function above for greater readbility ?